docs/modules/ROOT/pages/references/style-guide.adoc

```diff
@@ -0,0 +1,416 @@
+= Style guide
+
+The Jsonnet related parts of this guide build on top of https://github.com/databricks/jsonnet-style-guide/blob/30d4d54444c1001ba9561afd7c8de405630a2ca7/README.md[Jsonnet Guide] by https://databricks.com/[Databricks].
+
+http://creativecommons.org/licenses/by-nc-sa/4.0/[image:https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png[Creative Commons License]] +
+This work is licensed under a http://creativecommons.org/licenses/by-nc-sa/4.0/[Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License].
+
+== General style
+
+* Encode text files with UTF-8.
+* End lines with a single `\n` (lf).
+* Always end a file with a single `\n`.
+  This doesn't apply to empty files (for example `.gitkeep`).
+* Use two spaces for indentation except for `Makefile` where a single tab is to be used.
+* Trim whitespace from the end of a line.
+
+[NOTE]
+====
+Repositories of Project Syn provide an https://editorconfig.org/[EditorConfig].
+When using a https://editorconfig.org/#download[supported editor], space types and number of spaces will automatically be done correctly.
+====
+
+== Asciidoc style
+
+* Don't split sentences across lines but keep them on one single line.
+* Start a new line for each and every sentence.
+* Follow the https://docs.microsoft.com/en-us/style-guide/welcome/[Microsoft Writing Style Guide].
+
+[NOTE]
+====
+The Microsoft Writing Style is being enforced with https://github.com/errata-ai/vale[Vale].
+The implementation used can be found at https://github.com/errata-ai/Microsoft[errata-ai/Microsoft].
+====
+
+== Jsonnet style
+
+== Syntactic style
+
+=== Auto formatting
+
+Use `jsonnetfmt` with the `--pad-arrays` option to format files.
+This will fix basic style errors.
+
+[IMPORTANT]
+====
+Jsonnet is available in two implementations (https://github.com/google/jsonnet[jsonnet] and https://github.com/google/go-jsonnet[go-jsonnet]).
+Within the automated builds, we use the https://hub.docker.com/r/bitnami/jsonnet[Jsonnet docker image by Bitnami].
+Should there be any discrepancies between versions and flavours, the one packaged by Bitnami is the reference.
+====
+
+=== Variable declaration
+
+* Variables should be named in `camelCase` style, and should have self-explanatory names.
++
+[source,jsonnet]
+----
+local serverPort = 1000;
+local clientPort = 2000;
+----
+
+* Prefer `local` to `::` syntax for private/local variables.
+  Unlike `::`, variables defined with `local` can't be overridden by children, nor accessed by other files.
++
+[source,jsonnet]
+----
+{
+  // CORRECT
+  local myVariable = 3,
```

Comment:
Does this even work?

Reply:
Except from the  duplicated `result:`, this does work.

Reply:
I updated the example so it does work.